ruby/ext/google/protobuf_c/encode_decode.c

```diff
@@ -897,13 +897,29 @@ VALUE Message_decode(VALUE klass, VALUE data) {
  * format) under the interpretration given by this message class's definition
```

Comment:
Could you document the new params here please? Also please add docs for the hash params to encode_json (this should have been done before).

Reply:
Absolutely

Reply:
I wonder if it would be a little easier to follow if it was phrased more like this:

- ignore_unknown_fields: set true to ignore unknown fields (default is to raise an error)
- preserve_proto_fieldnames: set true to use original fieldnames (default is to camelCase)
- emit_defaults: set true to emit 0/false values (default is to omit them)

Reply:
totally works for me

Reply:
Updated. Thanks for the feedback.